packages/ember/tests/routing/decoupled_basic_test.js

```diff
@@ -0,0 +1,329 @@
+import { moduleFor, ApplicationTestCase } from 'internal-test-helpers';
+
+moduleFor('Basic Routing - Decoupled from global resovler',
+class extends ApplicationTestCase {
+  constructor() {
+    super();
+    this.addTemplate('home', '<h3 id="app">Hours</h3>');
+    this.addTemplate('camelot', '<section id="camelot"><h3>Is a silly place</h3></section>');
+    this.addTemplate('homepage', '<h3 id="troll">Megatroll</h3><p>{{model.home}}</p>');
+
+    this.router.map(function() {
+      this.route('home', { path: '/' });
+    });
+  }
+
+  getController(name) {
+    return this.applicationInstance.lookup(`controller:${name}`);
+  }
+
+  get currentPath() {
+    return this.getController('application').get('currentPath');
+  }
+
+  ['@test warn on URLs not included in the route set'](assert) {
+    return this.visit('/').then(() => {
+      expectAssertion(() => {
+        this.visit('/what-is-this-i-dont-even');
+      }, /'\/what-is-this-i-dont-even' did not match any routes/);
+    });
+  }
+
+  ['@test The Homepage'](assert) {
+    return this.visit('/').then(() => {
+      assert.equal(this.currentPath, 'home', 'currently on the home route');
+
+      let text = this.$('#app').text();
+      assert.equal(text, "Hours", 'the home template was rendered');
+    });
+  }
+
+  [`@test The Homepage and the Camelot page with multiple Router.map calls`](assert) {
+    this.router.map(function() {
+      this.route('camelot', { path: '/camelot' });
+    });
+
+    return this.visit('/camelot').then(() => {
+      assert.equal(this.currentPath, 'camelot');
+
+      let text = this.$('#camelot').text();
+      assert.equal(text, "Is a silly place",
+        'the camelot template was rendered'
+      );
+
+      return this.visit('/');
+    }).then(() => {
+      assert.equal(this.currentPath, 'home');
+
+      let text = this.$('#app').text();
+      assert.equal(text, "Hours", 'the home template was rendered');
+    });
+  }
+
+  [`@test The Homepage with explicity template name in renderTemplate`](assert) {
+    this.add('route:home', Route.extend({
+      renderTemplate() {
+        this.render('homepage');
+      }
+    }));
+
+    return this.visit('/').then(() => {
+      let text = this.$('#troll').text();
+      assert.equal(text, "Megatroll", 'the homepage template was rendered');
+    });
+  }
+
+  [`@test an alternate template will pull in an alternate controller`](assert) {
+    this.add('route:home',  Route.extend({
+      renderTemplate() {
+        this.render('homepage');
+      }
+    }));
+    this.add('controller:homepage', Controller.extend({
+      model: {
+        home: 'Comes from homepage'
+      }
+    }));
+
+    return this.visit('/').then(() => {
+      let text = this.$('p').text();
+
+      assert.equal(text, 'Comes from homepage',
+        'the homepage template was rendered'
+      );
+    });
+  }
+
+  [`@test An alternate template will pull in an alternate controller instead of controllername`](assert) {
```

Comment:
`controllerName`? Weird test title anyway, not sure why any variant of `controllername` needs to be in it.